### ISSUE TITLE
Nodes with same barcode number found

### ISSUE THREAD

**Issue by elhenriq** [precision-sustainable-ag/On-farm-Protocols#194]
I was going over the datasheet where we have the configuration string and notice there are a couple farms that have nodes with the same barcodes. 

18000612 is located in PF1 (17224903) & VT (280910)
18000616 is located in NH (280907) & VA east (280908)
18000617 is located in VA west (280913) &  MO (1084058)
18000621 is located in SC (1084035) & VA east (164668)
18000622 is located in KS (164670) & VA east (164668)
18000623 is located in IN (1084046) & INDIGO (280914)
18000624 is located in PFI (1724903) & FL (1084061)
18000625 is located in INDIGO (280912) & NE (1022432)
18000629 is located in INDIGO (1022433) & IN (164666)
18000630 is located in PFI (1724901) & MO (1084051)
18000831 is located in PFI (1724904) & VA east (1084047)
18000632 is located in VA east (1022429) & INDIGO (280914)
18000635 is located in INDIGO (280920) & NH (164678)
18000636 is located in NC (1084060) & INDIGO (280920) 

This is a significant amount of nodes that need to be dealt with. I recommend a fix on the database because it would take a while for us to send replacement to all this people. 

We started looking on the inventory, and we found that there is more than only these. From the looks of it, the duplicated barcodes start in 610 and extend to 636.  The ones I listed above are the only ones that have been deployed. We are collecting and separating the ones that  haven't been deployed yet, so we can come up with a solution.

**Comment by alondrathompson** [comment]
Oh wow. I'm going to forward this to Acclima. We can change the serial numbers, but we'd probably have to make them different enough so that we don't run into this issue the next time we order more.

**Comment by elhenriq** [comment]
yeah, finding the one from a couple of weeks ago between indigo and VA left me thinking there could be more.

**Comment by brianwdavis** [comment]
This is no good. There's no way to "fix it in the database" without knowing all the details of the device IDs each node is attached to and at what times (which is what the sensor install form is doing for us, but it relies on the serial numbers). The list you've provided is enough to get started, but the times are going to require a lot of hand-checking. This is going to be pretty messy.

**Comment by alondrathompson** [comment]
Has Ryan sent his node back so we can further investigate what's going on? I wonder if there's a mismatch somewhere between the barcode, the number on the label, what the barcode scanner on Kobo reads, and what's saved in the node memory.

**Comment by elhenriq** [comment]
found some more that are also duplicates. 

18000615 is located in IN (1084037) & on storage
18000619 is located in INDIGO (1084054) & on storage
18000620 is located in SC (280929) & on storage

there might be some more in there.

**Comment by elhenriq** [comment]
@brianwdavis is there a way to restrict the way the water data is plotted? like, force it to only plot the data for a specific farm if that node is associated to a specific gateway.

**Comment by brianwdavis** [comment]
Yes, that's how it already is. Unfortunately, the only way we currently have to identify the specific piece of hardware is the serial number in the database. I can use the linkage through the device ID to reassign new, fake serial numbers to that data, which is the plan. I just need to know what exactly Acclima plans to do to replace the duplicates (what numbers). That's part of why I would like to throw out the old serial numbers altogether, so then there's no confusion in the future about which data came in when. That is, 

> 18000612 is located in PF1 (17224903) & VT (280910)

Would become nodes `18PFI612` and `180VT612` in the database, or just make them totally different, like `17000001` and `17000002`, or whatever. Instead of keeping one as `18000612` and just changing the other to a new one.

**Comment by elhenriq** [comment]
also found 
18000626 was located in VA2 (280921) and storage
18000633 was located in indigo (1022434) and storage

**Comment by saseehav** [comment]
@brianwdavis time to remove the old data from all these serial numbers and replace with the uploaded files from individual farm codes.

**Comment by saseehav** [comment]
@brianwdavis status of nodes with duplicate serial numbers?

**Comment by brianwdavis** [comment]
This has been resolved in the DB.
